src/nanoemoji/nanoemoji.py

```diff
@@ -199,14 +275,29 @@ def _glyf_ufo(ufo, color_glyphs):
             color_glyph.glyph_name,
             svg_units_to_font_units,
         )
-        for idx, (paint, path) in enumerate(color_glyph.as_painted_layers()):
-            glyph_layer = _layer(ufo, idx)
+        for painted_layer in color_glyph.as_painted_layers():
+            _glyph(color_glyph, painted_layer)
 
-            glyph = ufo.get(color_glyph.glyph_name)
-            glyph.width = ufo.info.unitsPerEm
 
-            pen = TransformPen(glyph.getPen(), svg_units_to_font_units)
-            skia_path(path).draw(pen)
+def _colr_paint(colr_version: int, paint: Paint, palette: Sequence[Color]):
+    # For COLRv0, paint is just the palette index
+    # For COLRv1, it's a data structure describing paint
+    if colr_version == 0:
+        # COLRv0: draw using the first available color on the glyph_layer
+        # Results for gradients will be suboptimal :)
+        color = next(paint.colors())
+        return palette.index(color)
+
+    elif colr_version == 1:
+        # COLRv1: solid or gradient paint
+        return paint.to_ufo_paint(palette)
+
+    else:
+        raise ValueError(f"Unsupported COLR version: {colr_version}")
+
+
+def _reuse_key(painted_layer: PaintedLayer):
+    return (painted_layer.path.d, painted_layer.reuses)
```

Comment:
It would be better if all the reuse stuff was cleanly isolated to it's own section of the code which could have a nice overview comment.

For now renamed and added comments.